core/dbt/contracts/graph/manifest.py

```diff
@@ -898,10 +899,14 @@ def merge_from_artifact(
         refables = set(NodeType.refable())
         merged = set()
         for unique_id, node in other.nodes.items():
+            current = self.nodes[unique_id]
```

Comment:
is there a scenario where `unique_id` is present in the _other_ manifest, but not in _this_ manifest? I'm thinking that could happen if a model was deleted, for instance. If I'm thinking about that right, then I think this could raise a KeyError, and we'd probably want to use `self.nodes.get(unique_id)` and do something smart if the result is `None`. 

I might be way off-base here, it just jumped out at me though and wanted to confirm

Reply:
really good point, I was able to produce a `KeyError` here and changed the line to handle. this also gave me a clue to #2875 :)